sdk/storage/Azure.Storage.Blobs/src/AppendBlobClient.cs

```diff
@@ -1403,6 +1507,11 @@ public new AppendBlobClient WithEncryptionScope(string encryptionScope)
                     scope.Start();
                     ResponseWithHeaders<AppendBlobAppendBlockFromUrlHeaders> response;
 
+                    if (sourceBearerToken != null)
+                    {
+                        sourceBearerToken = $"Bearer {sourceBearerToken}";
```

Comment:
Ah. here it is. I don't think we should be doing this. If rest api expects this to be auth header we should just expose `SourceAuthorization` or something like this in options. See https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#authentication_schemes .
I wonder if for example Basic auth would just work with this.

Reply:
Maybe we should take https://docs.microsoft.com/en-us/dotnet/api/system.net.http.headers.authenticationheadervalue?view=net-5.0 rather than string?

Reply:
Good idea, implemented.